shared/js/dataDriver.js

```diff
@@ -85,6 +85,18 @@ loop.DataDriver = function() {
     }
 
     /**
+      * Add a page to the room's table of contents.
+      *
+      * @param {String} url       The url of the page.
+      * @param {String} title     The title of the page.
+      * @param {Object} metadata  Additional metadata,
```

Comment:
The single object allows the source of the object, e.g., `PageStore`, to add additional fields. I don't know what the structure will be, but I'm not sure if it's useful to have `DataDriver`'s API encode that. E.g., what if something should move out of `metadata` for some reason? Perhaps the image is important enough, so then we need to expand this API?

And the comment about nesting doesn't quite make sense as a nested object is being created right now for it to be stored in Firebase.

With the single object, `PageStore` can just send over its existing data structure of a page. E.g., some `textChat` objects happen to also have fields for `contentType` and `extraData` that `DataDriver` doesn't even need to be concerned about because there's some abstraction of what a `chat` object needs.